# Problem Statement

Visit With Us is a travel company that wants to identify customers who are likely to purchase a newly introduced wellness tourism package. Currently, the company relies on manual identification of potential customers, which is inefficient and inconsistent.

The objective of this project is to build a machine learning model that predicts whether a customer will purchase the wellness package. The solution will be integrated into an MLOps pipeline that includes data ingestion, preprocessing, model training, evaluation, deployment, and CI/CD automation using GitHub Actions and Hugging Face.

# Data Registration

Dateset has been created in the hugging face as



|
|
visit_with_us
|
|-data|
      |-train.csv

## Loading DataSet from Hugging Face

In [7]:
from datasets import load_dataset

dataset = load_dataset("gsri24/visit-with-us-wellness-data")
df = dataset["train"].to_pandas()

#### 🔭 Observation
> The dataset was successfully loaded from Hugging Face dataset repository. It contains customer demographic data i.e. demographic and interaction, to predict whether the customer is going to purchase the ~Wellness Package~.

## Data Overview

In [9]:
df.shape

(4128, 21)

#### 🔭 Observation:
> There are total of ~4128 rows~ and ~21 attributes~ in the dataset.

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4128 entries, 0 to 4127
Data columns (total 21 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Unnamed: 0                4128 non-null   int64  
 1   CustomerID                4128 non-null   int64  
 2   ProdTaken                 4128 non-null   int64  
 3   Age                       4128 non-null   float64
 4   TypeofContact             4128 non-null   object 
 5   CityTier                  4128 non-null   int64  
 6   DurationOfPitch           4128 non-null   float64
 7   Occupation                4128 non-null   object 
 8   Gender                    4128 non-null   object 
 9   NumberOfPersonVisiting    4128 non-null   int64  
 10  NumberOfFollowups         4128 non-null   float64
 11  ProductPitched            4128 non-null   object 
 12  PreferredPropertyStar     4128 non-null   float64
 13  MaritalStatus             4128 non-null   object 
 14  NumberOf

#### 🔭 Observation

The dataset has below columns

|Column Name| Description|
|---|---|
|CustomerID|Unique identifier for each customer.|
|ProdTaken|Target variable indicating whether the customer has purchased a package (0: No, 1: Yes).|
|Age|Age of the customer.|
|TypeofContact|The method by which the customer was contacted (Company Invited or Self Inquiry).|
|CityTier|The city category based on development, population, and living standards (Tier 1 > Tier 2 > Tier 3).
|Occupation|Customer's occupation (e.g., Salaried, Freelancer).
|Gender|Gender of the customer (Male, Female).
|NumberOfPersonVisiting|Total number of people accompanying the customer on the trip.
|PreferredPropertyStar|Preferred hotel rating by the customer.
|MaritalStatus|Marital status of the customer (Single, Married, Divorced).
|NumberOfTrips|Average number of trips the customer takes annually.
|Passport|Whether the customer holds a valid passport (0: No, 1: Yes).
|OwnCar|Whether the customer owns a car (0: No, 1: Yes).
|NumberOfChildrenVisiting|Number of children below age 5 accompanying the customer.
|Designation|Customer's designation in their current organization.
|MonthlyIncome|Gross monthly income of the customer.

In [12]:
df.describe().T

,count,mean,std,min,25%,50%,75%,max
Unnamed: 0,4128.0,2527.763808,1409.439133,0.0,1320.75,2603.5,3748.25,4887.0
CustomerID,4128.0,202527.763808,1409.439133,200000.0,201320.75,202603.5,203748.25,204887.0
ProdTaken,4128.0,0.193072,0.394757,0.0,0.00,0.0,0.00,1.0
Age,4128.0,37.231831,9.174521,18.0,31.00,36.0,43.00,61.0
CityTier,4128.0,1.663275,0.920640,1.0,1.00,1.0,3.00,3.0
DurationOfPitch,4128.0,15.584787,8.398142,5.0,9.00,14.0,20.00,127.0
NumberOfPersonVisiting,4128.0,2.949370,0.718818,1.0,2.00,3.0,3.00,5.0
NumberOfFollowups,4128.0,3.741521,1.006786,1.0,3.00,4.0,4.00,6.0
PreferredPropertyStar,4128.0,3.578488,0.795031,3.0,3.00,3.0,4.00,5.0
NumberOfTrips,4128.0,3.295300,1.856300,1.0,2.00,3.0,4.00,22.0


The below notebook will do the following steps
- remove customerId
- handling missing values
- encode categorical variables
- split the dataset

In [4]:
RANDOM_STATE = 97
TEST_SIZE = 0.2

In [5]:
from sklearn.model_selection import train_test_split


x = df.drop('ProdTaken', axis=1)
y = df['ProdTaken']

X_train, X_test, y_train, y_test = train_test_split(x, y, test_size=TEST_SIZE, random_state=RANDOM_STATE)